tools/tvk-oneclick/tvk-oneclick.sh

```diff
@@ -1018,60 +1071,74 @@ EOM
+      wait_install 15 "$cmd"
+      if ! kubectl get pods -l app=mysql-qa -n $backup_namespace 2>/dev/null | grep -q Running; then
+        echo "Application taking more time than usual to be in Ready state, Exiting.."
+        exit 1
+      fi
     else
       helm install mysql-qa stable/mysql -n $backup_namespace 1>> >(logit) 2>> >(logit)
       echo "Installing Application"
       cmd="kubectl get pods -l app=mysql-qa -n $backup_namespace 2>/dev/null | grep Running"
-      wait_install 10 "$cmd"
+      wait_install 15 "$cmd"
       if ! kubectl get pods -l app=mysql-qa -n $backup_namespace 2>/dev/null | grep -q Running; then
         echo "Application installation failed"
         return 1
       fi
       echo "Requested application is installed successfully"
-      yq eval -i 'del(.spec.backupPlanComponents)' backupplan.yaml 1>> >(logit) 2>> >(logit)
-      yq eval -i '.spec.backupPlanComponents.custom[0].matchLabels.app="mysql-qa"' backupplan.yaml 1>> >(logit) 2>> >(logit)
     fi
+    yq eval -i 'del(.spec.backupPlanComponents)' backupplan.yaml 1>> >(logit) 2>> >(logit)
+    yq eval -i '.spec.backupPlanComponents.custom[0].matchLabels.app="mysql-qa"' backupplan.yaml 1>> >(logit) 2>> >(logit)
     ;;
   2)
     if helm list -n $backup_namespace | grep -w -q my-wordpress 2>> >(logit); then
       echo "Application exists"
       if [[ "$if_resource_exists_still_proceed" != "Y" ]] && [[ "$if_resource_exists_still_proceed" != "y" ]]; then
         exit 1
       fi
+      echo "Waiting for application to be in Ready state"
     else
       #Add bitnami helm repo
       helm repo add bitnami https://charts.bitnami.com/bitnami 1>> >(logit) 2>> >(logit)
       helm install my-wordpress bitnami/wordpress -n $backup_namespace 1>> >(logit) 2>> >(logit)
       echo "Installing Application"
-      runtime=10
-      spin='-\|/'
-      i=0
-      endtime=$(python3 -c "import time;timeout = int(time.time()) + 60*$runtime;print(\"{0}\".format(timeout))")
-      while [[ $(python3 -c "import time;timeout = int(time.time());print(\"{0}\".format(timeout))") -le $endtime ]] && kubectl get pod -l app.kubernetes.io/instance=my-wordpress -n $backup_namespace -o jsonpath="{.items[*].status.conditions[*].status}" | grep -q False; do
-        i=$(((i + 1) % 4))
-        printf "\r %s" "${spin:$i:1}"
-        sleep .1
-      done
-      if kubectl get pod -l app.kubernetes.io/instance=my-wordpress -n $backup_namespace -o jsonpath="{.items[*].status.conditions[*].status}" | grep -q False; then
-        echo "Wordpress installation failed"
-        return 1
-      fi
-      echo "Requested application is installed successfully"
-      yq eval -i 'del(.spec.backupPlanComponents)' backupplan.yaml 1>> >(logit) 2>> >(logit)
     fi
+    runtime=20
+    spin='-\|/'
+    i=0
+    endtime=$(python3 -c "import time;timeout = int(time.time()) + 60*$runtime;print(\"{0}\".format(timeout))")
```

Comment:
why python command needed here?
timeout can be managed by bash commands

Reply:
Initially I tried with plain bash but the date functionality was breaking on MAC . 'date' syntax is different on MAC and Linux. That's why changed the code to support both MAC and Linux